subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java

```diff
@@ -50,6 +52,11 @@ public MavenArtifactRepository getRepository() {
      */
     public void setRepository(MavenArtifactRepository repository) {
         this.repository = repository;
+
+        Credentials credentials = ((AuthenticationSupportedInternal) repository).getConfiguredCredentials();
```

Comment:
Calling `getConfiguredCredentials()` here evaluates the credentials provider if any was set.
It so happens that the way current publishing tasks are wired makes this code run before any of the tasks are executed, effectively giving us a fail-fast behavior if the provider was set but its value can not be resolved. This is validated in the integration test.
The following input is thus never utilized.

I am not sure if this is the way to go to achieve this fail-fast behavior. If it is, then perhaps introducing a better named method to `ensureCredentialsPresent()`, or even `validateCredentials()` on the internal repository class would do.